extensions/arc/src/ui/dialogs/connectSqlDialog.ts

```diff
@@ -122,11 +157,37 @@ export abstract class ConnectToSqlDialog extends InitializingComponent {
 			return true;
 		}
 		else {
-			vscode.window.showErrorMessage(this.connectionFailedMessage(result.errorMessage));
-			return false;
+			// Show error with instructions for MSSQL Provider Encryption error code -2146893019 thrown by SqlClient when certificate validation fails.
+			if (result.errorCode === -2146893019) {
+				return this.showInstructionTextAsWarning(connectionProfile, async updatedConnection => {
+					return await this.connect(updatedConnection);
+				});
+			} else {
+				vscode.window.showErrorMessage(this.connectionFailedMessage(result.errorMessage));
+				return false;
+			}
 		}
 	}
 
+	private async showInstructionTextAsWarning(profile: azdata.IConnectionProfile, reconnectAction: IReconnectAction): Promise<boolean> {
+		const selection = await vscode.window.showWarningMessage(
```

Comment:
So currently this shows just the explanation text when it fails as a warning. It seems to me : 

1. That it should be an error (the connection failed after all)
2. It should tell the user that the connection failed (and the error message) so it's clear what happened in addition to the current explanation of why